src/Core/Framework/DataAbstractionLayer/Dbal/EntityReader.php

```diff
@@ -60,6 +60,8 @@ public function __construct(
     ) {
     }
 
+    private bool $partial = false;
```

Comment:
I am not sure, if we should introduce a state to the class :thinking: 

wouldn't it better to pass that down the methods?

Reply:
that's currently the case, but the passed through is just for the current nested association. 

The entity hydrator already has this "I'm in partial hydration mode". In this mode it always uses partial entities.

Reply:
I am just asking a question :blush: no request for changes (yet) :wink:

Reply:
@OliverSkroblin Thank you for checking this. And sorry for the delay.

Yes, this resolves the issue.